File: Java/src/test/java/kata/BankOcrTest.java
package kata;

import org.junit.Test;

import java.io.IOException;
import java.util.List;

import static kata.Input.allDigits;
import static org.junit.Assert.assertEquals;

public class BankOcrTest {

    @Test
    public void shouldParseOcrLine() throws IOException {
        // TODO failing test, start here
        // guiding test from outside in as starting point.
        // a) ignore this test
        // b) or stub everything in the BankOcr constructor
        // c) or let it be red until the end
        BankOcr bankOcr = new BankOcr();

        List<AccountNumber> accountNumbers = bankOcr.parse(allDigits());

        assertEquals(1, accountNumbers.size());
        assertEquals(new AccountNumber("123456789"), accountNumbers.get(0));
    }


}

File: Csharp/BankOcr.Kata.Tests/BankOcrTest.cs
using NUnit.Framework;

namespace Kata.Tests
{
    [TestFixture]
    public class BankOcrTest
    {
        [Test]
        public void ShouldParseOcrLine()
        {
            // TODO failing test, start here
            // guiding test from outside in as starting point.
            // a) ignore this test
            // b) or stub everything in the BankOcr constructor
            // c) or let it be red until the end

            var bankOcr = new BankOcr();

            var accountNumbers = bankOcr.Parse(Input.AllDigits());

            Assert.AreEqual(1, accountNumbers.Count);
            Assert.AreEqual(new AccountNumber("123456789"), accountNumbers[0]);
        }

    }
}
